### ISSUE TITLE
Fetch schema from Apollo Registry

### ISSUE THREAD

**Issue by nigelhanlon** [nearform/mercurius-apollo-registry#7]
Currently, the plugin only allows you to upload a local schema to the Apollo registry. Since the registry acts as a central repository and source of truth for the GraphQL schema, it would also be good if the plugin could download and apply a schema to the running Mercurius instance. 

**Suggestions and Considerations**

- The plugin should operate in either a reporting or consuming mode, allowing a given server to either report its local schema to the registry or download and apply the latest version from the registry.
- The [replace schema method](https://github.com/mercurius-js/mercurius/blob/master/docs/api/options.md#appgraphqlreplaceschemaschema) of Mercurius may be of interest here.
- We need to investigate the possibility of a given schema not working with the local Mercurius instance (eg resolvers missing etc).
- It is not clear if there is a notification mechanism we can subscribe to for schema changes in the registry, so for now, some kind of polling may be needed.
- It is not clear how to actually fetch the schema from the registry but there are CLI tools for Apollo that may give some insight. https://github.com/apollographql/apollo-tooling#apollo-servicedownload-output

**Comment by mcollina** [comment]
I don't really understand what the benefit of this would be... I don't think we should replace the schema without replacing the code that handles it.

The only scenario I see this working is with federation: we could pull the schema and target URL for the services to mesh together.

**Comment by simoneb** [comment]
This is correct @mcollina. The whole idea behind having a registry is to use it in a Federated scenario. Otherwise, as you mention, if the service which publishes the schema is the same that consumes it, there's not much of a point, since the schema must be backed by a corresponding implementation. Therefore, the idea is that the services behind a Federated gateway each publish their schema to the registry, while the gateway reads the schema from the registry instead of building it at runtime from the underlying services.

[This Netflix post](https://netflixtechblog.com/how-netflix-scales-its-api-with-graphql-federation-part-2-bbe71aaec44a) explains  this well and the picture below is clarifying. I'm sure you're familiar with this, I'm adding this for reference.

If your question is whether reading from the registry should be in this plugin or not, that's something we haven't discussed yet but we thought it would make sense to track as an issue.

![image](https://user-images.githubusercontent.com/20181/107435620-ad96b400-6b2c-11eb-8660-c8a7bbfbc349.png)
